### ISSUE TITLE
Pylance deletes automatically-added `#` sign when you line break inside a comment

### ISSUE THREAD

**Issue by fireattack** [microsoft/pylance-release#5484]
<h3> Environment data </h3>

-   Language Server version: 2024.2.2
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): Python 3.12
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: off

<h3> Code Snippet </h3>

```python
# this is a comment.
```

<h3> Repro Steps </h3>

1. In any python file, find a comment.
2. In the middle of the comment, press "Enter".

<h3> Expected behavior </h3>

A `# ` is automatically added to the new (second) line.

<h3> Actual behavior </h3>

The `# ` is indeed added, but immediately got deleted by Pylance. You can even press Ctrl+Z to revert the deletion of `# `.

I know it's caused by Pylance because if I disable it (but keep the python extension enabled), the bug is gone.

This is a recent regression; it used to work fine.

There is also an occasional indentation bug as mentioned https://github.com/microsoft/pylance-release/issues/5488, which came with this bug together, so the root cause is likely the same.

<h3> Logs </h3>

```
2024-02-19 15:01:44.873 [info] (1072) [FG] parsing: file:///c%3A/Users/ikena/Desktop/vsocdeportabletest/aaa.py (1ms)
2024-02-19 15:01:44.874 [info] (1072) [FG] binding: file:///c%3A/Users/ikena/Desktop/vsocdeportabletest/aaa.py (0ms)
2024-02-19 15:01:45.167 [info] (1072) [BG(1)] analyzing: file:///c%3A/Users/ikena/Desktop/vsocdeportabletest/aaa.py ...
2024-02-19 15:01:45.167 [info] (1072) [BG(1)]   parsing: file:///c%3A/Users/ikena/Desktop/vsocdeportabletest/aaa.py (1ms)
2024-02-19 15:01:45.167 [info] (1072) [BG(1)]   binding: file:///c%3A/Users/ikena/Desktop/vsocdeportabletest/aaa.py (0ms)
2024-02-19 15:01:45.201 [info] (1072) [BG(1)]   checking: file:///c%3A/Users/ikena/Desktop/vsocdeportabletest/aaa.py ...
2024-02-19 15:01:45.201 [info] (1072) [BG(1)]     parsing: file:///c%3A/Users/ikena/Desktop/VSCode-win32-x64-1.86.2/data/extensions/ms-python.vscode-pylance-2024.2.2/dist/typeshed-fallback/stdlib/typing.pyi [fs read 1ms] (34ms)
2024-02-19 15:01:45.206 [info] (1072) [FG] parsing: file:///c%3A/Users/ikena/Desktop/vsocdeportabletest/aaa.py (1ms)
2024-02-19 15:01:45.207 [info] (1072) [FG] binding: file:///c%3A/Users/ikena/Desktop/vsocdeportabletest/aaa.py (0ms)
2024-02-19 15:01:45.214 [info] (1072) [BG(1)]     binding: file:///c%3A/Users/ikena/Desktop/VSCode-win32-x64-1.86.2/data/extensions/ms-python.vscode-pylance-2024.2.2/dist/typeshed-fallback/stdlib/typing.pyi (14ms)
2024-02-19 15:01:45.235 [info] (1072) [BG(1)]     parsing: file:///c%3A/Users/ikena/Desktop/VSCode-win32-x64-1.86.2/data/extensions/ms-python.vscode-pylance-2024.2.2/dist/typeshed-fallback/stdlib/types.pyi [fs read 0ms] (16ms)
2024-02-19 15:01:45.246 [info] (1072) [BG(1)]     binding: file:///c%3A/Users/ikena/Desktop/VSCode-win32-x64-1.86.2/data/extensions/ms-python.vscode-pylance-2024.2.2/dist/typeshed-fallback/stdlib/types.pyi (9ms)
2024-02-19 15:01:45.270 [info] (1072) [BG(1)]     parsing: file:///c%3A/Users/ikena/Desktop/VSCode-win32-x64-1.86.2/data/extensions/ms-python.vscode-pylance-2024.2.2/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 1ms] (11ms)
2024-02-19 15:01:45.273 [info] (1072) [BG(1)]     binding: file:///c%3A/Users/ikena/Desktop/VSCode-win32-x64-1.86.2/data/extensions/ms-python.vscode-pylance-2024.2.2/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (4ms)
2024-02-19 15:01:45.284 [info] (1072) [BG(1)]     parsing: file:///c%3A/Users/ikena/Desktop/VSCode-win32-x64-1.86.2/data/extensions/ms-python.vscode-pylance-2024.2.2/dist/typeshed-fallback/stdlib/abc.pyi [fs read 0ms] (3ms)
2024-02-19 15:01:45.284 [info] (1072) [BG(1)]     binding: file:///c%3A/Users/ikena/Desktop/VSCode-win32-x64-1.86.2/data/extensions/ms-python.vscode-pylance-2024.2.2/dist/typeshed-fallback/stdlib/abc.pyi (1ms)
2024-02-19 15:01:45.287 [info] (1072) [BG(1)]   checking: file:///c%3A/Users/ikena/Desktop/vsocdeportabletest/aaa.py (121ms)
2024-02-19 15:01:45.288 [info] (1072) [BG(1)] analyzing: file:///c%3A/Users/ikena/Desktop/vsocdeportabletest/aaa.py (122ms)
2024-02-19 15:01:45.292 [info] (1072) [BG(1)] indexing: file:///c%3A/Users/ikena/Desktop/vsocdeportabletest/aaa.py [found 0] (1ms)
2024-02-19 15:01:45.292 [info] (1072) Indexing Done: file:///c%3A/Users/ikena/Desktop/vsocdeportabletest/aaa.py
2024-02-19 15:01:45.509 [info] (1072) Background analysis message: getSemanticTokens delta
2024-02-19 15:01:45.509 [info] (1072) [BG(1)] getSemanticTokens delta previousResultId:1708326090091 at file:///c%3A/Users/ikena/Desktop/vsocdeportabletest/aaa.py (1ms)

```

**Comment by fireattack** [comment]
I bisected the regression window:

* First bad: 2024.1.100
* Last good: 2023.12.104

**Comment by debonte** [comment]
> This is a recent regression; it used to work fine.

This has been broken since our auto-indent system was switched over to use `formatOnType` in 2022.9.21. I can't explain why this appears to be a recent regression on your machine.

Note that when trying to repro this in older builds:
- Before 2023.4.31, you need to have the auto-indent experiment enabled.
- You need to use a compatible build of the Python extension (that knows about the experiment)

I used:
- Python: 2022.14.0
- Pylance: 2022.9.21
- Settings:
```json
    "[python]": {
        "editor.formatOnType": true
    },
    "editor.formatOnType": true,
    "python.experiments.optInto": ["pylanceAutoIndent"],
```

**Comment by debonte** [comment]
Maybe when this bug did not repro for you, you had `editor.formatOnType` set to `false`?

In any case, while waiting for this to be fixed, you could work around this issue by disabling our formatOnType-based auto-indent support by putting the following in your `settings.json`:

```json
"[python]":{
    "editor.formatOnType": false
}
```

**Comment by fireattack** [comment]
Yeah, my editor.formatOnType is off, which is the default I think (I used a totally clean user profile for the testing.)

And indeed, I can reproduce the bug with Python: 2022.14.0 + Pylance: 2022.9.30 (can't find .21) + settings you gave above.

**Comment by fireattack** [comment]
Ok, that's weird, I can't reproduce "good" result on 2023.12.104 anymore.. 

So no idea what happened, but at this point I think it doesn't really matter since you have already confirmed the bug.

**Comment by debonte** [comment]
> maybe Pylance only started modifying default value of [python].editor.formatOnType at this point? 

Yes, when Pylance is installed, the default value of `editor.formatOnType` for python is `true`. Our extension `package.json` changes the default value. This change was made in 2023.4.31 when the auto-indent feature was enabled for all users (i.e. when the experiment was removed).

In any case, I appreciate you reporting this. We'll get it fixed. It's just a bit less urgent now that we understand that it didn't regress recently.

**Comment by KacieKK** [comment]
This issue has been fixed in prerelease version 2024.5.103, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20245103-28-May-2024-PreRelease)
